d3a_interface/data_classes.py

```diff
@@ -326,14 +365,19 @@ def to_json_string(self) -> str:
         return json.dumps(trade_dict, default=json_datetime_serializer)
 
     @classmethod
-    def from_json(cls, trade_string, current_time=None) -> "Trade":
+    def from_json(cls, trade_string) -> "Trade":
+        """De-serialize trade from json string."""
         trade_dict = json.loads(trade_string)
-        trade_dict["offer_bid"] = BaseBidOffer.from_json(trade_dict["offer_bid"], current_time)
-        if key_in_dict_and_not_none(trade_dict, "residual"):
-            trade_dict["residual"] = BaseBidOffer.from_json(trade_dict["residual"],
-                                                            current_time)
-        trade_dict["creation_time"] = parse(trade_dict["creation_time"])
-        if key_in_dict_and_not_none(trade_dict, "offer_bid_trade_info"):
+        trade_dict["offer_bid"] = BaseBidOffer.from_json(trade_dict["offer_bid"])
+        if trade_dict.get("residual"):
+            trade_dict["residual"] = BaseBidOffer.from_json(trade_dict["residual"])
+        if trade_dict.get("creation_time"):
+            trade_dict["creation_time"] = str_to_pendulum_datetime(trade_dict["creation_time"])
+        else:
+            trade_dict["creation_time"] = None
+        if trade_dict.get("time_slot"):
+            trade_dict["time_slot"] = str_to_pendulum_datetime(trade_dict[""])
```

Comment:
What is `trade_dict[""]`?

Reply:
Maybe this was meant to be the following?
```suggestion
            trade_dict["time_slot"] = str_to_pendulum_datetime(trade_dict["time_slot"])
```

Reply:
Wow, nice catch! Let me fix it and add a unittest.